common/src/main/java/org/apache/uniffle/common/compression/SnappyCodec.java

```diff
@@ -64,4 +64,21 @@ public byte[] compress(byte[] src) {
       throw new RssException("Failed to uncompress by Snappy", e);
     }
   }
+
+  @Override
+  public int compress(ByteBuffer src, ByteBuffer dest) {
```

Comment:
Could we use `ByteBuffer.duplicate` method to reduce the modification of source Bytebuffer. We shouldn't modify the source ByteBuffer's position and limit. WDTY @advancedxy

Reply:
It's ok. consistent with `ChecksumUtils.getCrc32`